server/channels/app/channel.go

```diff
@@ -1239,6 +1239,10 @@ func (a *App) UpdateChannelMemberSchemeRoles(c request.CTX, channelID string, us
 		return nil, model.NewAppError("UpdateChannelMemberSchemeRoles", "api.channel.update_channel_member_roles.user_and_guest.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if !isSchemeUser {
+		return nil, model.NewAppError("UpdateChannelMemberSchemeRoles", "api.channel.update_channel_member_roles.unset_user_scheme.app_error", nil, "", http.StatusBadRequest)
+	}
+
```

Comment:
I'd like to understand this change better, mostly because it makes this method even weirder: it accept an `isSchemeUser` and an `isSchemeGuest`, but you're not allowed to actually set those values to be true.

What's going on under the covers?